File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.java
class Solution {
  public int numberOfBeams(String[] bank) {
    int prev=0,ans=0;
    for(var s: bank) {     //var==auto//
      int cnt=0;
      for(var i=0;i<s.length();i++)
        if(s.charAt(i)=='1') {
          cnt++;
        }
      if(cnt>0) {
        ans+=prev*cnt;
        prev=cnt;
      }
    }
    return ans;
  }
}


// class Solution {
//   public int numberOfBeams(String[] bank) {
//     int prev = 0, ans = 0;

//     for (String s: bank) {
//       int count = 0;
//       for (int i = 0; i < s.length(); i++)
//         if (s.charAt(i) == '1') {
//           count++;
//         }

//       if (count > 0) {
//         ans += prev * count;
//         prev = count;
//       }
//     }

//     return ans;
//   }
// }

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int start=0;
        int ans=0;
        for(auto& i : bank)
        {
            int curr=count(i.begin(),i.end(),'1');
            if(curr>0)
            {
                ans+=start*curr;
                start=curr;
            }
        }
        return ans;
    }
};


// class Solution {
// public:
//     int numberOfBeams(vector<string>& bank) {
//         int prev = 0, ans = 0;
        
//         for (string s : bank) {
//             int count = 0;
//             for (char c : s) {
//                 if (c == '1') {
//                     count++;
//                 }
//             }
//             if (count != 0) {
//                 ans += (prev * count);
//                 prev = count;
//             }
//         }
        
//         return ans;
//     }
// };